keptn/keptn
# feat: Introduce signed Keptn Helm charts

### This PR
- adds functionality to sign helm charts with a private/public keypair
- signing only happens for releases and pre-releases
- the public key is added to the repo and to every new release together with the Helm provenance `*.tgz.prov` files
- adds a small section to the readme to point users into the right direction if they want to verify the keptn charts

#### How to test
Check the helm chart artifacts in this run for the provenance files: https://github.com/keptn/keptn/actions/runs/2919304598
They can be verified by downloading the public key under `assets/pubring.gpg` and importing it into gpg.
Use the following steps:
- Download the helm chart and provenance file from the workflow run, as well as the public key. Then run below commands.
- `gpg --import pubring.gpg`
- `helm verify keptn-0.19.0-dev-PR-8730.tgz`

(you will maybe need to export your gpg public key ring since Helm cannot handle `*.kbx` keyrings, only `*.gpg` keyrings, more info on this can be found [here](https://helm.sh/docs/topics/provenance/#:~:text=Warning%3A%20the,gnupg/secring.gpg))

Fixes #6610
